src/vcpkg/commands.build.cpp

```diff
@@ -1702,115 +1702,113 @@ namespace vcpkg
+                output
+                    .append_raw(
+                        fmt::format("\\e[0Ksection_start:{}:{}[collapsed=true]\r\\e[0K", timestamp, section_name))
+                    .append_raw(title)
+                    .append_raw('\n')
+                    .append_raw(contents)
+                    .append_raw(fmt::format("\\e[0Ksection_end:{}:{}\r\\e[0K\n", timestamp, section_name));
+            }
+            break;
+            case CIKind::AzurePipelines:
+                // https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#formatting-commands
+                output.append_raw("##vso[task.uploadfile]")
+                    .append_raw(file)
+                    .append_raw('\n')
+                    .append_raw("##[group]")
+                    .append_raw(title)
+                    .append_raw('\n')
+                    .append_raw(contents)
+                    .append_raw("##[endgroup]\n");
+                break;
         }
     }
 
     LocalizedString create_user_troubleshooting_message(const InstallPlanAction& action,
+                                                        CIKind detected_ci,
                                                         const VcpkgPaths& paths,
                                                         const std::vector<std::string>& error_logs,
-                                                        const Optional<Path>& issue_body)
+                                                        const Optional<Path>& maybe_issue_body)
     {
         const auto& spec_name = action.spec.name();
         const auto& triplet_name = action.spec.triplet().to_string();
         LocalizedString result = msg::format(msgBuildTroubleshootingMessage1).append_raw('\n');
         result.append_indent().append_raw(make_gh_issue_search_url(spec_name)).append_raw('\n');
         result.append(msgBuildTroubleshootingMessage2).append_raw('\n');
 
-        if (issue_body.has_value())
+        if (auto issue_body = maybe_issue_body.get())
         {
-            auto ci = detect_ci_type();
-            const auto path = issue_body.get()->generic_u8string();
+            auto& fs = paths.get_filesystem();
+            const auto path = issue_body->generic_u8string();
+            const bool collapsible = is_collapsible_ci_kind(detected_ci);
             const auto body =
-                ci.map([&](auto) {
-                      return fmt::format("Copy issue body from collapsed section \"{}\" in the ci log output",
-                                         issue_body.get()->filename());
-                  }).value_or(Strings::concat("Copy issue body from ", path));
+                collapsible ? msg::format(msgCopyIssueBodyFromCollapsibleSection, msg::path = issue_body->filename())
+                            : msg::format(msgCopyIssueBodyFromFile, msg::path = path);
```

Comment:
I did not localize these messages since it breaks the automatic handling of issues in the `vcpkg` repo: 
https://github.com/microsoft/vcpkg/blob/96dbba1e03d1ecc7c7096e97d5a09ff3c58e21df/.github/workflows/check_issues.yml#L49
https://github.com/microsoft/vcpkg/blob/96dbba1e03d1ecc7c7096e97d5a09ff3c58e21df/.github/workflows/check_issues.yml#L80

But localizing the text maybe improves the rate of correctly submitted issues because the users understand the text and the check could probably be adapted to the new messages.

Reply:
As far as I understand it this is explaining to the user how to make such an issue, not the content of the issue itself?

Reply:
It is the content of the issue itself

Reply:
Hmmmm... OK, commented and delocalized that part.